mne/io/tests/test_apply_function.py

```diff
@@ -62,3 +68,18 @@ def test_apply_function_verbose():
         assert out is raw
         raw.apply_function(printer, verbose=True)
         assert sio.getvalue().count("\n") == n_chan
+
+
+def test_apply_function_ch_access():
+    """Test apply function is able to access channel idx."""
+    n_chan = 2
+    n_times = 1
+    ch_names = [str(ii) for ii in range(n_chan)]
+    d = np.zeros((n_chan, n_times))
+    for ch_idx in range(n_chan):
+        d[ch_idx] += ch_idx
+    raw = RawArray(d, create_info(ch_names, 1.0, "mag"))
```

Comment:
Let's make the `raw` creation simpler, the 2 lines below create a Raw object with the channels `'0'` and `'1'`, which hold 0 and 1 respectively in their channel data, during 10 samples.

```suggestion
    data = np.full((2, 10), np.arange(2).reshape(-1, 1))
    raw = RawArray(data, create_info(2, 1.0, "mag"))
```